### PR TITLE
[R] Added ApiClient and fixed other issues

### PR DESCRIPTION
**Pull request by brnleehng**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [X] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [X] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [X] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming langauge.

### Description of the PR

Super excited to see a R swagger code generator! I've added a couple of changes and now have a fully working swagger client. 

1) Added **ApiClient.R**: Following the Python code generator and other languages, a HTTP client handler adds flexibility for users to build on top of the default swagger code if user needs such as special authorization handling. 

2) Changed Api operators and how they are called. Some parameters are required while others parameters are not. However, the current API operators forces you to have all of the parameters.

3) Fixed toJSON / fromJSON: The sprintf does not work if the values are NULL. Switched it to use lists instead of strings for flexbility
-  Object$new()$fromJSON : Can't call fromJSON from a new object

4) Added auto generated documentation in the mustache templates. It's somewhat ambiguous on how to use the models and operators, hopefully this would help. 

5) Fixed the Description file to use appName. However it doesn't completely solve the issue of generating a R file if appName has spaces - Then it will fail. 

6) Added Response and Element classes in R

### CODE REVIEW

File: modules/swagger-codegen/src/main/resources/r/api.mustache [hunk 1]
```diff
@@ -1,56 +1,94 @@
...
+#' }
+#' 
+#' @export
 {{classname}} <- R6::R6Class(
   '{{classname}}',
   public = list(
     userAgent = "{{#httpUserAgent}}{{{.}}}{{/httpUserAgent}}{{^httpUserAgent}}Swagger-Codegen/{{{packageVersion}}}/r{{/httpUserAgent}}",
-    basePath = "{{{basePath}}}",
-    initialize = function(basePath){
-      if (!missing(basePath)) {
```
Comment (wing328) [inline modules/swagger-codegen/src/main/resources/r/api.mustache]: @brnleehng thanks for the PR. Does it mean developers (users of the SDK) can no longer use a different base path during runtime when constructing the class?

Comment (wing328) [reply modules/swagger-codegen/src/main/resources/r/api.mustache]: Ah please ignore. You've moved "basePath" to apiClient.

**Comment by brnleehng** [comment]
@wing328 Hmm... I'm not sure why I'm seeing this error. I don't think any of my changes would affect the travis CI.

**Comment by wing328** [comment]
@brnleehng the Travis CI (MacOS image) times out from time to time so please kindly ignore and we'll restart the job before the merge.

**Comment by wing328** [comment]
Not for this PR but we probably want to add some tests (syntax checker, Petstore integration tests, etc) as we're seeing more changes to enhance the R generator.

**Pull request merged**